src/Services/TokenRequestor.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pinnacle\OpenIdConnect\Services;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\Psr7\Uri;
+use GuzzleHttp\RequestOptions;
+use GuzzleHttp\Utils;
+use InvalidArgumentException;
+use Pinnacle\OpenIdConnect\Exceptions\AccessTokenNotFoundException;
+use Pinnacle\OpenIdConnect\Models\Contracts\ProviderConfigurationInterface;
+use Pinnacle\OpenIdConnect\Exceptions\OpenIdConnectException;
+use Psr\Log\LoggerInterface;
+use stdClass;
+
+class TokenRequestor
+{
+    public function __construct(
+        private ProviderConfigurationInterface $provider,
+        private Uri                            $redirectUri,
+        private string                         $codeVerifier,
+        private ?LoggerInterface               $logger = null
+    ) {
+    }
```

Comment:
I was opting to leave this class untouched as much as possible since it will be reworked quite a bit in the next PR. But I can make these changes now if you think it's necessary

Reply:
It's not necessary, no. We'll likely have to cut a new major version after the re-work anyway. I was mostly just trying to avoid BC-breaks.